DFE-Digital/sts-plan-technology-for-your-school
# Feat: support defaults and null references to fix content missing relationships

## Overview

Addresses ticket [#220759](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_workitems/edit/220759)

Adds defaults to int/bool/string types coming in from Contentful and makes any content references nullable. 
This is to ensure the database always saves changes coming from contentful, regardless of completeness, to prevent the issue where relationships sometimes don't form depending on the order that content is created 

## Changes

### Major

- Sections can have a null interstitial page
- Warnings can have null text contents
- TextBodies can have null text contents
- All required fields are now defaulted to empty string, 0, or false, depending on their type, if not present in a contentful web message

### Minor

- Updated the seeded test database content to include examples of content with these missing references, that can be used to check the page loads without error when `usePreview=false` 

## How to review the PR

1. Follow the steps in the `SeedTestData` project readme to spin up a mock database. 
2. Change your dotnet secrets connection string to point at the mock database
3. Run plan tech web against it and make sure the pages all load without error

Note that you may need to sign out and back in for the users to start working properly

Separately, it would be worthwhile to disable the Azure function in dev, run it locally, and add content in contentful with required fields missing, and make sure that it always comes through

## Additional Notes

This would ideally have e2e tests but needs the seeded test db to be setup with these firsts - so this will be documented on that ticket as test cases to include.

## Checklist
